sktime/forecasting/trend/_pwl_trend_forecaster.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements a piecewise linear trend forecaster by wrapping fbprophet."""
+
+__author__ = ["sbuse"]
+
+import pandas as pd
+
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.base.adapters import _ProphetAdapter
+
+
+class PiecewiseLinearTrendForecaster(_ProphetAdapter):
+    """
+    Forecast time series data with a piecwise linear trend.
+
+    The forecaster uses Facebook's prophet algorithm [1]_ and extracts the piecewise
+    linear trend from it. Only hyper-parameters relevant for the trend modelling are
+    exposed via the constructor.
+
+    Data can be passed in one of the sktime compatible formats,
+    naming a column `ds` such as in the prophet package is not necessary.
+
+    Unlike vanilla `prophet`, also supports integer/range and period index:
+    * integer/range index is interpreted as days since Jan 1, 2000
+    * `PeriodIndex` is converted using the `pandas` method `to_timestamp`
+
+    Parameters
+    ----------
+    changepoints: list or None, default=None
+        List of dates at which to include potential changepoints. If
+        not specified, potential changepoints are selected automatically.
+    n_changepoints: int, default=25
+        Number of potential changepoints to include. Not used
+        if input `changepoints` is supplied. If `changepoints` is not supplied,
+        then n_changepoints potential changepoints are selected uniformly from
+        the first `changepoint_range` proportion of the history.
+    changepoint_range: float, default=0.8
+        Proportion of history in which trend changepoints will
+        be estimated. Defaults to 0.8 for the first 80%. Not used if
+        `changepoints` is specified.
+    changepoint_prior_scale: float, default=0.05
+        Parameter modulating the flexibility of the
+        automatic changepoint selection. Large values will allow many
+        changepoints, small values will allow few changepoints.
+        Recommended to take values within [0.001,0.5].
+
+    References
+    ----------
+    .. [1] https://facebook.github.io/prophet
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
```

Comment:
doctests are always run, even if prophet is not present, so we should skip it - see with the `Prophet` forecaster how this is done.

There is an idiomatic `doctest` way which I do not remember - and, currently, skipif, and move the code to a`pytest` test is currently the most common in `sktime`.